src/foam/core/NullService.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+
+/**
+ * NullService facilitates the creation of null object of a given interface.
+ *
+ * Thus removing the need to manually create class for the null implementation
+ * of the service interface eg. NullMyService.
+ *
+ * See. NullServiceTest.java for usage.
+ *
+ */
+public class NullService {
```

Comment:
How often will this be useful? Even for Null services don't you need to know what the null operation is? For example, a validator interface would still need to know if the default NOP was to return true or false.

Reply:
If the interface is implemented with default operations like these, then you could do the same by just implementing the interface (and extending more easily where needed).

Reply:
This is at runtime, we can spin off a null service on the fly without the need to add code, re-compile, re-build and re-deploy.

Might not be useful for a null service that we want to control what the output is. Can be handy for creating null service for interfaces having more void (no return value) methods.